GetF!t/Modules/Exercises/ExerciseView.swift

```diff
@@ -0,0 +1,102 @@
+import SwiftUI
+import AVKit
+
+struct ExerciseView: View {
+    @ObservedObject var exerciseViewModel: ExercisesViewModel
+    @EnvironmentObject var history: HistoryViewModel
+    @EnvironmentObject var selectedTabManager: SelectedTabManager
+
+    var body: some View {
+        GeometryReader { geometry in
+            VStack {
+                HeaderView(headerViewModel: HeaderViewModel.init(), titleText: exerciseViewModel.exercise)
+                    .padding(.bottom)
+                Spacer()
+                ContainerView {
+                    VStack {
+                        video(size: geometry.size)
+                        startExerciseButton
+                        Spacer()
+                        RatingView(ratingViewModel: RatingViewModel.init(exerciseIndex: exerciseViewModel.index))
+                            .padding()
+                        historyButton
+                    }
+                }
+                .frame(height: geometry.size.height * 0.8)
+                .sheet(
+                    isPresented: $exerciseViewModel.showSheet,
+                    onDismiss: exerciseViewModel.onDismissLogic
+                ) {
+                    switchLogic
+                }
+            }
+           // .onChange(of: exerciseViewModel.shouldAddExercise) {
+           //     if $0 {
+           //         history.addDoneExercise(Exercise.exercises[exerciseViewModel.index].exerciseName)
+           //     }
+           // }
+            .onAppear {
+                exerciseViewModel.addDoneExercise = history.addDoneExercise
+                exerciseViewModel.goToNextTab = selectedTabManager.goToNextTab
+            }
+        }
+    }
+
+    @ViewBuilder
+    var switchLogic: some View {
+        if let exerciseSheet = exerciseViewModel.exerciseSheet, exerciseViewModel.indexLimit {
+            switch exerciseSheet {
+            case .history:
+                HistoryView()
+            case .timer:
+                TimerView()
+            case .success:
+                SuccessView()
+            }
+        }
+    }
+
+    @ViewBuilder
+    private func video(size: CGSize) -> some View {
+        if exerciseViewModel.indexLimit {
+            if let url = Bundle.main.url(
+                forResource: exerciseViewModel.video,
+                withExtension: "mp4") {
+                VideoPlayer(player: AVPlayer(url: url))
+                    .frame(height: size.height * 0.25)
+                    .clipShape(RoundedRectangle(cornerRadius: 12))
+                    .padding(20)
+            } else {
+                Text(LocalizedStringProvider.Errors.couldntFind)
+                    .foregroundColor(.red)
+            }
```

Comment:
do viewModelu treba pridať computed property, ktorá ti vráti optional `URL?` tu vo view potom spraviť len:
```swift
if let url = viewModel.videoURL {
    VideoPlayer(...)
} else {
    Text(...)
}
```

Reply:
a na ten indexlimit by som sa vykašľala, aj tak ti celá appka spadne ak tam dáš nejaké číslo s väčším indexom ako je počet cvičení 😄

Reply:
o tom indexu jsem se tenkrat bavil s Macem a rikal ze je to v cajku kdyz tam dam podminku